processor/attributesprocessor/attributes.go

```diff
@@ -125,3 +151,110 @@ func setAttribute(action attributeAction, attributesMap map[string]*tracepb.Attr
 		attributesMap[action.Key] = value
 	}
 }
+
+// shouldProcessSpan determines if a span should be processed.
+// True is returned when a span should be processed.
+// False is returned when a span should not be processed.
+// The logic determining if a span should be processed is set
+// in the attribute configuration with the include and exclude settings.
+// Include properties are checked before exclude settings are checked.
+func (a *attributesProcessor) shouldProcessSpan(span *tracepb.Span, serviceName string) bool {
+	// By default all spans are processed no include and exclude properties are set.
+	if a.config.include == nil && a.config.exclude == nil {
+		return true
+	}
+
+	if a.config.include != nil {
+		// A false returned in this case means the span should not be processed.
+		if include := matchSpanToProperties(*a.config.include, span, serviceName); !include {
+			return false
+		}
+	}
+
+	if a.config.exclude != nil {
+		// A true returned in this case means the span should not be processed.
+		if exclude := matchSpanToProperties(*a.config.exclude, span, serviceName); exclude {
+			return false
+		}
+	}
+
+	return true
+}
+
+// matchProperties
+// There are two sets of properties to match against.
+// The service name is checked first, if specified. The attributes are checked
+// afterwards, if specified.
+// At least one of services or attributes must be specified. It is supported
+// to have both specified, but both `services` and `attributes` must evaluate
+// to true for a match to occur.
+func matchSpanToProperties(mp matchingProperties, span *tracepb.Span, serviceName string) bool {
+
+	// Check if the service name is in the list of service names.
+	serviceFound := mp.Services[serviceName]
+	// Only return false only when all of these conditions are true:
+	// - a valid service name is provided. Invalid names are empty string,
+	// `<nil-batch-node>`, `<nil-service-info>` and `<empty-service-info-name>`.
+	// The last three values are from ServiceNameForNode method.
+	// - the map isn't empty (when the map is empty, serviceFound is always false.)
+	// - serviceFound is false
+	if serviceName != "" &&
```

Comment:
Ahh - I have a bug here. I don't need to check what the value is of serviceName because by doing that it means the service name check isn't exact and we have cases that by pass it. I'm going to remove this. 

ex the current code (with the bug) 
Trace
```
ServiceName: ""
```

Matching Properties Include
```
Services: ["svcA", "svcB"]
```
The above trace would match which isn't what we want. 

Fixing this and adding the missing test case for it.